rerun_cpp/CMakeLists.txt

```diff
@@ -168,18 +174,20 @@ if(RERUN_DOWNLOAD_AND_BUILD_ARROW)
         GIT_PROGRESS true
         CMAKE_ARGS
         --preset ninja-debug-minimal
-        -DARROW_IPC=ON
+        -DARROW_BOOST_USE_SHARED=OFF
         -DARROW_BUILD_SHARED=${ARROW_BUILD_SHARED}
         -DARROW_BUILD_STATIC=${ARROW_BUILD_STATIC}
-        -DCMAKE_INSTALL_PREFIX=${ARROW_DOWNLOAD_PATH}
+        -DARROW_CXXFLAGS=${DARROW_CXXFLAGS}
+        -DARROW_IPC=ON
+        -DARROW_JEMALLOC=OFF
+        -DARROW_MIMALLOC=ON
         -DARROW_USE_ASAN=OFF
         -DARROW_USE_TSAN=OFF
         -DARROW_USE_UBSAN=OFF
-        -DARROW_JEMALLOC=OFF
-        -Dxsimd_SOURCE=BUNDLED
         -DBOOST_SOURCE=BUNDLED
-        -DARROW_BOOST_USE_SHARED=OFF
-        -DARROW_CXXFLAGS=${DARROW_CXXFLAGS}
+        -DCMAKE_BUILD_TYPE=${ARROW_CMAKE_BUILD_TYPE}
+        -DCMAKE_INSTALL_PREFIX=${ARROW_DOWNLOAD_PATH}
+        -Dxsimd_SOURCE=BUNDLED
```

Comment:
We should sort these defines if their order doesn't matter

Reply:
This PR sorts them.